R/coh_plot.R

```diff
@@ -0,0 +1,324 @@
+    must.include = c(outcome_status_col, time_to_event_col, vacc_status_col)
+  )
+  checkmate::assert_character(
+    c(vaccinated_color, unvaccinated_color)
+  )
+  checkmate::assert_logical(
+    percentage, len = 1
+  )
+  checkmate::assert_logical(
+    cumulative, len = 1
+  )
+  # check date types
+  checkmate::assert_date(
+    start_cohort, any.missing = FALSE, len = 1
+  )
+  checkmate::assert_date(
+    end_cohort, any.missing = FALSE, len = 1
+  )
+  checkmate::assert_names(
+    data[[vacc_status_col]],
+    must.include = c(vaccinated_status, unvaccinated_status)
+  )
+
+  #KM model
+  km_model <- survival::survfit(
+    survival::Surv(
+      data[[time_to_event_col]],
+      data[[outcome_status_col]]
+    ) ~ data[[vacc_status_col]]
+  )
+
+  #Extract data from survival element
+  results <- extract_surv_model(km_model, start_cohort, end_cohort)
+
+  if (cumulative) {
+    results$plot <- 1 - results$surv
+    results$plot_lower <- 1 - results$upper
+    results$plot_upper <- 1 - results$lower
+  } else {
+    results$plot <- results$surv
+    results$plot_lower <- results$upper
+    results$plot_upper <- results$lower
+  }
+
+  #Filter data by status
+  conn1 <- paste0("data[[vacc_status_col]]=", vaccinated_status)
+  results_1 <- results[results$strata == conn1, ]
+  conn2 <- paste0("data[[vacc_status_col]]=", unvaccinated_status)
+  results_2 <- results[results$strata == conn2, ]
+
+  #Plot
+  colors <- c("c1" = vaccinated_color, "c2" = unvaccinated_color)
+  vacc_status <- c("c1" = vaccinated_status, "c2" = unvaccinated_status)
+  plt <- ggplot2::ggplot() +
+    ggplot2::geom_ribbon(
+      ggplot2::aes(x = results_1$time,
+        ymin = results_1$plot_lower, ymax = results_1$plot_upper
+      ),
+      fill = colors[1], alpha = 0.2
+    ) +
+    ggplot2::geom_step(
+      ggplot2::aes(x = results_1$time, y = results_1$plot,
+        color = "c1"
+      )
+    ) +
+    ggplot2::geom_ribbon(
+      ggplot2::aes(x = results_2$time,
+        ymin = results_2$plot_lower, ymax = results_2$plot_upper
+      ),
+      fill = colors[2], alpha = 0.2
+    ) +
+    ggplot2::geom_step(
+      ggplot2::aes(x = results_2$time, y = results_2$plot,
+        color = "c2"
+      )
+    ) + {
+    if (percentage) {
+      ggplot2::scale_y_continuous(labels = scales::percent)
+    }
+  } +
+    ggplot2::scale_color_manual(name = "Vaccine Status",
+      values = colors,
+      labels = vacc_status
+    ) +
+    ggplot2::theme_classic() + {
+    if (cumulative) {
+      ggplot2::ylab("Cumulative hazard")
+    } else {
+      ggplot2::ylab("Survival probability")
+    }
+  } +
+    ggplot2::xlab("Time to event (Days)") +
+    ggplot2::labs(colour = "Vaccine Status") +
+    ggplot2::theme(axis.text = ggplot2::element_text(size = 13),
+                   axis.title = ggplot2::element_text(size = 15),
+                   legend.title = ggplot2::element_text(size = 15),
+                   legend.text = ggplot2::element_text(size = 13))
```

Comment:
I would leave these to the default values and remove this chunk, as setting fixed values leads to bad scaling when the output figure size changes. Users can add these details if they feel it's necessary later.

If you want to call attention to certain plot text, can I suggest using bold or italics instead?

Reply:
I removed this. For a future PR I will work on a plot theme for vaccineff. See solution in https://github.com/epiverse-trace/vaccineff/pull/31/commits/7f2096caada259d2de1cbfe9d0717dcebadf1a31